### ISSUE TITLE
alert = driver().switch_to.alert alert.accept() throws NoAlertPresentException

### ISSUE THREAD

**Issue by byakatat** [yashaka/selene#128]
When i use selene and try to accept alert:
```
config.browser_name = Browser.CHROME

alert = driver().switch_to.alert
alert.accept()
```
**I get this error:**
..\..\..\MobilePages\RecipientPageM.py:23: in choose_recipient_from_list
    alert.accept()
..\..\..\..\..\selene-virt\lib\site-packages\selenium\webdriver\common\alert.py:80: in accept
    self.driver.execute(Command.ACCEPT_ALERT)
..\..\..\..\..\selene-virt\lib\site-packages\selenium\webdriver\remote\webdriver.py:233: in execute
    self.error_handler.check_response(response)
**E       selenium.common.exceptions.NoAlertPresentException: Message: no alert open**
E         (Session info: chrome=58.0.3029.81)
E         (Driver info: chromedriver=2.29.461591 

While when i use standart driver it works fine:
```
self.driver = webdriver.Chrome(ChromeDriverManager().install())

alert = self.driver.switch_to.alert
alert.accept()
```

**Comment by SergeyPirogov** [comment]
@byakatat are you sure that alert apears? Selene don't wait for alert to appear

**Comment by byakatat** [comment]
@SergeyPirogov I'm absolutely sure. 
I even tried to put time.sleep(10) before switch to alert, but it also doesn't work. Alert appears and after 10 sec i got this error.
As I said when i use standard driver it works fine without any waitings for alert to present.

**Comment by SergeyPirogov** [comment]
And this is simple web application? Not Angular or something else?

**Comment by byakatat** [comment]
@SergeyPirogov yes, it's simple web app. Here is a screenshot of alert https://yadi.sk/i/qi-MXW3V3HTHQg

**Comment by yashaka** [comment]
will look into it...

**Comment by yashaka** [comment]
Hm... We have tests for alerts: https://github.com/yashaka/selene/blob/master/tests/integration/test_alert.py

And they pass... That means that something may be wrong with your alert, @byakatat :)

Can you share link to production site? So we can look more closely what kind of alert do you have?

**Comment by byakatat** [comment]
@yashaka  
Sure. Use Chrome browser.

1) Login here [login page](https://m.florist.ru/users/login/)    
with email: tsz73303@tipsb.com and password: 123456
2) Click on the logo on the left top corner - a main page will be opened
3) Click on any bouquet - a bouquet page will be opened
4) Click "Оформить заказ" button - a cart page will be opened
5) Click "Оформить заказ" button again - a delivery date page will be opened
6) Click "Следующий шаг" button - a recipient page will be opened
7) And here you should choose recipient from dropdown, look here https://yadi.sk/i/pAdyELYD3JC9Sm
and allert will appear.

And as I mentioned, when I use standard driver tests pass.

**Comment by yashaka** [comment]
Thanks, I will try to find the reason...

**Comment by aleksandr-kotlyar** [comment]
### Check

I have checked the situation in actual 2.0.34a selene core api version. And it seems no longer actual.

I resurrected alert tests for 2.0.34a core api: 8b53e719c632ab308401bf52a3745c9f7cbd698c and ensured that they work fine.

```python
import os
import pytest
from selenium import webdriver
from selenium.common.exceptions import NoAlertPresentException
from webdriver_manager.chrome import ChromeDriverManager
from selene import Config, Browser

start_page = 'file://' + os.path.abspath(os.path.dirname(__file__)) + '/../resources/start_page.html'


@pytest.fixture(scope='session')
def session_driver():
    chrome_driver = webdriver.Chrome(ChromeDriverManager().install())
    yield chrome_driver
    chrome_driver.quit()


@pytest.fixture(scope='function')
def session_browser(session_driver):
    yield Browser(Config(driver=session_driver))


def test_can_accept_alert(session_browser):
    session_browser.open(start_page)

    session_browser.element("#alert_btn").click()
    session_browser.driver.switch_to.alert.accept()

    try:
        session_browser.driver.switch_to.alert
        assert False, 'actual: alert presents, expected: alert not present'
    except NoAlertPresentException:
        assert True


def test_can_dismiss_confirm_dialog(session_browser):
    session_browser.open(start_page)

    session_browser.element("#alert_btn").click()
    session_browser.driver.switch_to.alert.dismiss()

    try:
        session_browser.driver.switch_to.alert
        assert False, 'actual: alert presents, expected: alert not present'
    except NoAlertPresentException:
        assert True


def test_alert_is_present(session_browser):
    session_browser.open(start_page)

    session_browser.element("#alert_btn").click()

    try:
        session_browser.driver.switch_to.alert
        assert True
    except NoAlertPresentException:
        assert False, 'actual: alert not present, expected: alert is present'
```

### Suggestion

The problem of TC could be in selene version 1.0.10a because it hadn't clear and consistent api yet. There is no code to check he used selene api properly.

I upped tests of selene version 1.0.10a too, and had the trouble with the api, cause 

```python
from selene.browser import driver, visit

visit(page)
driver().switch_to.alert.accept()
```

opened new window instead of shared driver when i call `driver()`.

So the script of calling alert in selene 1.0.10a should look like in tests below:

```python
import os
from time import sleep

from selenium.common.exceptions import NoAlertPresentException

from selene import factory, browser

start_page = 'file://' + os.path.abspath(os.path.dirname(__file__)) + '/../resources/start_page.html'


# todo: ensure works and enabled
def test_can_accept_alert():
    browser.visit(start_page)

    browser.element("#alert_btn").click()
    shared_driver = factory.get_shared_driver()
    shared_driver.switch_to.alert.accept()

    try:
        shared_driver.switch_to.alert.accept()
        assert False, 'actual: alert presents, expected: alert not present'
    except NoAlertPresentException:
        assert True


def test_can_dismiss_confirm_dialog():
    browser.visit(start_page)

    browser.element("#alert_btn").click()
    shared_driver = factory.get_shared_driver()
    shared_driver.switch_to.alert.dismiss()

    try:
        shared_driver.switch_to.alert.dismiss()
        assert False, 'actual: alert presents, expected: alert not present'
    except NoAlertPresentException:
        assert True
```

### Resolution
Issue not reproduced. The problem could be in not proper usage of the selene - check previous "Suggestion" block above.

Issue can be closed until we look up the whole code the TC will give to reproduce the issue.

**Comment by aleksandr-kotlyar** [comment]
Seems it could be this bug https://github.com/yashaka/selene/issues/117.
This bug has been fixed in https://github.com/yashaka/selene/pull/118.

**Comment by aleksandr-kotlyar** [comment]
@byakatat does this problem still actual on `selene v2.0.0a34`? Can you provide more details please, because UI from your example has been changed and that's why i cannot reproduce your case. But we have selene tests which shows up it's working now and i found PR which should fix this problem.

> Seems it could be this bug, the same issue at the same time #117.
> This bug has been fixed in PR #118.

**Comment by yashaka** [comment]
So, after all discussions... This seems to be not a bug. 

Yet, the discussion revealed the topic of "waiting for alert". I have created a separate issue for that: #248
